extensions/MozillaIAM/template/en/default/hook/admin/users/userdata-end.html.tmpl

```diff
@@ -0,0 +1,36 @@
+[%# This Source Code Form is subject to the terms of the Mozilla Public
+  # License, v. 2.0. If a copy of the MPL was not distributed with this
+  # file, You can obtain one at http://mozilla.org/MPL/2.0/.
+  #
+  # This Source Code Form is "Incompatible With Secondary Licenses", as
+  # defined by the Mozilla Public License, v. 2.0.
+  #%]
+
+[%
+  RETURN UNLESS Param('mozilla_iam_enabled');
+  RETURN UNLESS otheruser.iam_username;
+  moz_emp_group_id = 0;
+  FOREACH group = groups;
+    IF group.name == 'mozilla-employee-confidential';
+      moz_emp_group_id = group.id;
+      LAST;
+    END;
+  END;
+  RETURN IF !moz_emp_group_id;
+%]
+
+<script [% script_nonce FILTER none %]>
+window.addEventListener('DOMContentLoaded', () => {
+    const label = document.querySelector('label[for="group_[% moz_emp_group_id FILTER js %]"]');
+    if (label) {
+      const span  = document.createElement('span');
+      span.setAttribute('title',
+        'Changes made by admins will be overwritten next ' +
+        'time the user\'s People profile is updated.');
+      span.innerHTML = '(IAM Managed)';
+      label.appendChild(span);
+    }
+    document.querySelector('input[name="group_[% moz_emp_group_id FILTER js %]"]').disabled = true;
+    document.querySelector('input[name="bless_[% moz_emp_group_id FILTER js %]"]').disabled = true;
+});
+</script>
```

Comment:
Can we do this we render the user settings page, rather than modifying it afterwards with javascript?
This will have a couple benefits:
1. People who have javascript disabled will have a better experience
2. It will be easier to follow where work happens: a future developer won't be stuck looking at the user settings page template, wondering how some inputs appear to be being disabled.

Reply:
It comes down to how extensions work in Bugzilla. MozillaIAM is an extension and not part of the core code. The template that generates the user permissions admin page do not have extension hooks in the places that render each row of the permissions table. There is only the hook that is called at the end of the table so that was the reason for using jscript to update the DOM after render. I can do it the other way, fixing it pre-render, but will take adding additional extension hooks. 

The code is supposed to be a mild deterrent really to keep future admins from messing with that particular group setting and is not meant to be bullet-proof really. If someone wants to update the group, they can just blank out the value for IAM Username, save, and then edit the permission. The think is that the next time the user logs in with Mozilla IAM, it will repopulate the username and set the permission anyway.

Reply:
Ah, that makes sense, I didn't realize the extension-y model going on here.
This is ok then :+1:  :)